src/embedding/udf.rs

```diff
@@ -0,0 +1,143 @@
+use std::time::Duration;
+
+use crate::datatype::vecf32::{Vecf32, Vecf32Output};
+use crate::gucs::embedding::{openai_options, OpenAIOptions};
+use crate::prelude::embedding_failed;
+use pgrx::default;
+use reqwest::blocking::Client;
+use service::prelude::F32;
+
+use super::openai::{EmbeddingRequest, EmbeddingResponse};
+
+#[pgrx::pg_extern(volatile, strict)]
+fn _vectors_ai_embedding_vector(
+    input: String,
+    model: default!(String, "'text-embedding-ada-002'"),
```

Comment:
A model may be not popular or depcrated. It's not very good to set default value for `model`.

Reply:
This is widely used as default: https://github.com/search?q=text-embedding-ada-002&type=code
As a stright string, it easy to maintain when it is depcrated, and it's important for user to input as less as he could.

Reply:
`text-embedding-3-small` should be the new default. It's 5x cheaper than 002 and has better performance

Reply:
Someone who uses model `abc` could always create a new function named `text2vec_openai_abc` or `my_text2vec` to call `text2vec_openai`.

Reply:
We can add `text2vec_openai_ada002` and `text2vec_openai_3small`.

Reply:
> We can add `text2vec_openai_ada002` and `text2vec_openai_3small`. (I think it is a user's duty but it's not bad if we provide it.)

LGTM. The problem of default model name is that there are many embedding providers nowadays such as jina/cohere/voyage. They have different model names.